utils/modeler/modeler.go

```diff
@@ -0,0 +1,107 @@
+package modeler
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+// Modeler is a utility for populating an arbitrary model's fields with values from a
+// map[string]string.
+type Modeler struct {
+	prefix string
+	tag    string
+}
+
+// NewModeler returns a pointer to a new Modeler, confgiured with the provided map key prefix and
+// struct tag key.
+func NewModeler(prefix string, tag string) *Modeler {
+	return &Modeler{prefix: prefix, tag: tag}
+}
+
+// MapToModel populates the provided model with values from the provided map.
+func (m *Modeler) MapToModel(data map[string]string, out interface{}) error {
+	rv := reflect.ValueOf(out)
+	return m.mapToModel(data, "", rv)
+}
+
+func (m *Modeler) mapToModel(data map[string]string, context string, rv reflect.Value) error {
+	// If rv is invalid (represents a nil litersl), we cannot proceed.
+	if rv.Kind() == reflect.Invalid {
+		return fmt.Errorf("Cannot populate a nil from the map.")
```

Comment:
Can you make this and the below errors exported, so that you can document some of the expected errors that will be returned by the above `MapToModel` func? The errors with variable information in them can be `struct`s that adhere to [the `error` interface](https://godoc.org/builtin#error)